walletlibrary/src/main/java/com/microsoft/walletlibrary/requests/requirements/AccessTokenRequirement.kt

```diff
@@ -29,12 +31,15 @@ class AccessTokenRequirement(
     override val required: Boolean = false,
 
     internal var accessToken: String? = null
-): Requirement {
+) : Requirement {
     // Validates the requirement and throws an exception if the requirement is invalid or not fulfilled.
-    override fun validate(): Result<Unit> {
+    override fun validate(): VerifiedIdResult<Unit> {
         if (accessToken == null)
-            return Result.failure(AccessTokenRequirementNotFulfilledException("AccessTokenRequirement has not been fulfilled."))
-        return Result.success(Unit)
+            return RequirementNotMetException(
+                "Access Token has not been set.",
+                VerifiedIdExceptions.REQUIREMENT_NOT_MET_EXCEPTION.value
+            ).toVerifiedIdResult()
+        return VerifiedIdResult.success(Unit)
```

Comment:
same here. would using the getResult { } function be cleaner?

Reply:
for all Validate functions below?

Reply:
Haven't used getResult{} in validate functions since getResult{} is suspend function and any function using it have to be suspend as well. But validate functions don't need to be suspend.